### ISSUE TITLE
InllegalStateException while recovering directive

### ISSUE THREAD

**Issue by la10736** [intellij-rust/intellij-rust#1032]
```
Worker exited due to exception: Check failed.
java.lang.IllegalStateException: Check failed.
	at org.rust.ide.docs.RsDocumentationProvider.formatSignature(RsDocumentationProvider.kt:83)
	at org.rust.ide.docs.RsDocumentationProvider.getQuickNavigateInfo(RsDocumentationProvider.kt:99)
	at org.rust.ide.docs.RsDocumentationProvider.getQuickNavigateInfo(RsDocumentationProvider.kt:28)
	at com.intellij.lang.documentation.CompositeDocumentationProvider.getQuickNavigateInfo(CompositeDocumentationProvider.java:120)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler.generateInfo(CtrlMouseHandler.java:325)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler.access$1900(CtrlMouseHandler.java:105)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$InfoSingle.getInfo(CtrlMouseHandler.java:447)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$TooltipProvider.doExecute(CtrlMouseHandler.java:838)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$TooltipProvider.access$2900(CtrlMouseHandler.java:776)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$TooltipProvider$1.performInReadAction(CtrlMouseHandler.java:821)
	at com.intellij.openapi.progress.util.ReadTask$1.compute(ReadTask.java:70)
	at com.intellij.openapi.progress.util.ReadTask$1.compute(ReadTask.java:67)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:899)
	at com.intellij.openapi.progress.util.ReadTask.runBackgroundProcess(ReadTask.java:67)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runUnderProgress$4(ProgressIndicatorUtils.java:221)
	at com.intellij.openapi.progress.impl.CoreProgressManager$4.run(CoreProgressManager.java:199)
	at com.intellij.openapi.progress.impl.CoreProgressManager$3.run(CoreProgressManager.java:179)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:568)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:519)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:196)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runUnderProgress(ProgressIndicatorUtils.java:219)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.access$000(ProgressIndicatorUtils.java:44)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils$4.run(ProgressIndicatorUtils.java:179)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

What I was doing is a rollback (CTRL-z) that resume follow code at the head of file:

```
#![feature(plugin)]
#![plugin(mockers_macros)]
```

I hope it is useful...

**Comment by alygin** [comment]
It's fixed in the pending PR #1020. There are a couple of cases that cause such exception. I think I fixed all of them.

**Comment by la10736** [comment]
Ok... thx

**Comment by matklad** [comment]
closed in #1020, alpha published.
